helm/cluster-api-ipam-provider-in-cluster/templates/pools/pool.yaml

```diff
@@ -1,22 +1,33 @@
 {{- if .Values.enabled }}
 {{- with .Values.inClusterIpPool }}
 {{- if .enabled }}
-apiVersion: ipam.cluster.x-k8s.io/v1alpha1
-kind: InClusterIPPool
+apiVersion: v1
+kind: ConfigMap
 metadata:
-  name: {{ .name }}
-spec:
-  gateway: 0.0.0.0
-  prefix: 24
-{{- if .start }}
-  start: {{ .start }}
-  end: {{ .end }}
-{{- else }}
-  addresses:
-  {{- range .addresses }}
-  - {{ . }}
-  {{- end }}
-{{- end }}
+  name: in-cluster-ip-pool
+  namespace: {{ $.Release.Namespace }}
+  annotations:
+    helm.sh/hook: "post-install,post-upgrade"
+    helm.sh/hook-delete-policy: "before-hook-creation,hook-succeeded,hook-failed"
+    helm.sh/hook-weight: "-2"
+data:
+  pool.yaml: |-
+    apiVersion: ipam.cluster.x-k8s.io/v1alpha1
+    kind: InClusterIPPool
```

Comment:
It would be better to fetch these manifests from another folder so that we can manage them easily. See https://github.com/giantswarm/cilium-app/blob/main/helm/cilium/templates/default-policies/configmap.yaml#L9